backend/app/core/permissions.py

```
@@ -125,42 +125,24 @@ async def check_permissions(
-            return await func(*args, **kwargs)
-        return wrapper
-    return decorator
+def require_permissions_dependency(required_permissions: List[Permission]):
+    """Dependency to enforce permission checks for FastAPI routes."""
+    async def inner(current_user: Users = Depends(get_current_user), db: AsyncSession = Depends(get_db)) -> None:
+        await check_permissions(required_permissions, current_user, db)
+    return inner
+
+def require_any_permissions_dependency(required_permissions: List[Permission]):
+    """Dependency that allows access if user has ANY of the required permissions."""
+    async def inner(current_user: Users = Depends(get_current_user), db: AsyncSession = Depends(get_db)) -> None:
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)